File: WPF/MVVM/Converters/AlphaConverter.vb
Imports System.Globalization
Imports System.Windows.Data
Imports System.Windows.Media

Namespace Common.MVVM.Converters

    Public Class AlphaConverter
        Implements IValueConverter

        Public Function Convert(ByVal Value As Object, ByVal TargetType As Type, ByVal Parameter As Object, ByVal Culture As CultureInfo) As Object Implements IValueConverter.Convert
            Dim B = TryCast(Value, SolidColorBrush)
            Dim C As Color
            If B IsNot Nothing Then
                C = B.Color
            Else
                C = CType(Value, Color)
            End If
            Dim P = CByte(255)
            If Parameter IsNot Nothing Then
                P = CType(Parameter, Byte)
            End If
            C = Color.FromArgb(P, C.R, C.G, C.B)
            If TargetType = GetType(Color) Then
                Return C
            End If
            Return New SolidColorBrush(C)
        End Function

        Public Function ConvertBack(ByVal Value As Object, ByVal TargetType As Type, ByVal Parameter As Object, ByVal Culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
            Dim B = TryCast(Value, SolidColorBrush)
            Dim C As Color
            If B IsNot Nothing Then
                C = B.Color
            Else
                C = CType(Value, Color)
            End If
            Dim P = CByte(255)
            If Parameter IsNot Nothing Then
                P = CType(Parameter, Byte)
            End If
            C = Color.FromArgb(P, C.R, C.G, C.B)
            If TargetType = GetType(Color) Then
                Return C
            End If
            Return New SolidColorBrush(C)
        End Function

    End Class

End Namespace

File: WPF/MVVM/Converters/AlphaConverter.cs
using System;
using System.Globalization;
using System.Windows.Data;
using System.Windows.Media;

namespace Ks.Common.MVVM.Converters
{
    public class AlphaConverter : IValueConverter
    {
        public object Convert(object Value, Type TargetType, object Parameter, CultureInfo Culture)
        {
            var B = Value as SolidColorBrush;
            Color C;
            if (B != null)
            {
                C = B.Color;
            }
            else
            {
                C = (Color) Value;
            }

            var P = (byte) 255;
            if (Parameter != null)
            {
                P = System.Convert.ToByte(Parameter);
            }

            C = Color.FromArgb(P, C.R, C.G, C.B);
            if (TargetType == typeof(Color))
            {
                return C;
            }

            return new SolidColorBrush(C);
        }

        public object ConvertBack(object Value, Type TargetType, object Parameter, CultureInfo Culture)
        {
            var B = Value as SolidColorBrush;
            Color C;
            if (B != null)
            {
                C = B.Color;
            }
            else
            {
                C = (Color) Value;
            }

            var P = (byte) 255;
            if (Parameter != null)
            {
                P = System.Convert.ToByte(Parameter);
            }

            C = Color.FromArgb(P, C.R, C.G, C.B);
            if (TargetType == typeof(Color))
            {
                return C;
            }

            return new SolidColorBrush(C);
        }
    }
}
